Submissions/Laguerta_Week3.playground/Contents.swift

```diff
@@ -0,0 +1,217 @@
+}
+greet("John", "Red Curry")
+
+//: 2.)
+//:
+//: Rewrite the closure to return zero for all odd numbers.
+//:
+var numbers = [20, 19, 7, 12]
+
+//: full "syntactic sugar" form:
+var closureNumbers = numbers.map({ (number: Int) -> Int in
+    let result = 0
+    if number % 2 != 0 {
+        return result
+    } else {
+        return number
+    }
+})
+print(closureNumbers)
+//: omitting parameters and return type form:
+let mappedNumbers = numbers.map({ number in number % 2 != 0 ? 0: number})
+print(mappedNumbers)
+//: reference variable form:
+let mappedNums = numbers.map { $0 % 2 != 0 ? 0: $0}
+print(mappedNums)
+//: ## Objects and Classes
+//: 3.)
+//:
+//: Add a constant property with `let`, and add another method that takes an argument.
+//:
+//: 4.)
+//:
+//: This version of the `Shape` class is missing something important: an initializer to set up the class when an instance is created. Use `init` to create one.
+//:
+class Shape {
+    let definition = "A 'shape' is simply a geometric figure."
+    var numberOfSides = 0
+    
+    init(definition: String, numberOfSides: Int) {
+        self.numberOfSides = numberOfSides
+    }
+    
+    func simpleDescription() -> String {
+        return "A shape with \(numberOfSides) sides."
+    }
+    func areaOfQuadrilaterals(base: Float, height: Float) -> Float{
+        return (base * height)
+    }
+    
+}
+
+//: 5.)
+//:
+//: Make a subclass of `NamedShape` called `Circle` that takes a radius and a name as arguments to its initializer. Implement an `area()` and a `simpleDescription()` method on the `Circle` class.
+//:
+class NamedShape {
+    var numberOfSides: Int = 0
+    var name: String
+    
+    init(name: String) {
+        self.name = name
+    }
+    
+    func simpleDescription() -> String {
+        return "A shape with \(numberOfSides) sides."
+    }
+}
+
+class Circle: NamedShape {
+    var radius: Float = 3.0
+    
+    init(radius: Float, name: String){
+        self.radius = radius
+        super.init(name: name)
+    }
+    
+    func area() -> Float {
+        
+        return radius * radius * 3.14
+    }
+    
+    override func simpleDescription() -> String {
+        return "A shape with no sides."
+    }
+}
+
+
+//: ## Enumerations and Structures
+//: 6.)
+//:
+//: Write a function that compares two `Rank` values by comparing their raw values.
+//:
+enum Rank: Int, CaseIterable {
+    case ace = 1
+    case two, three, four, five, six, seven, eight, nine, ten
+    case jack, queen, king
+    
+    func simpleDescription() -> String {
+        switch self {
+        case .ace:
+            return "ace"
+        case .jack:
+            return "jack"
+        case .queen:
+            return "queen"
+        case .king:
+            return "king"
+        default:
+            return String(self.rawValue)
+        }
+    }
+
+}
+let ace = Rank.ace
+let aceRawValue = ace.rawValue
+
+func compareRank(rank1: Rank, rank2: Rank) -> String {
```

Comment:
I like how thorough your function is. Can you write the declaration so it's more readable at the call site, and I'll know the kind of thing it might print?